### PR TITLE
Add deletePodsOp to scheduler_perf

### PR DESCRIPTION
**Pull request by macsko**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

We need to be able to begin pod deletion a bit later, especially for queueing hint performance tests. That's why a separate op seems reasonable. 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/integration/scheduler_perf/scheduler_perf.go [hunk 1]
```diff
@@ -665,6 +661,44 @@ func (cpso createPodSetsOp) patchParams(w *workload) (realOp, error) {
...
+// deletePodsOp defines an op where previously created pods are deleted.
+// The test can block on the completion of this op before moving forward or
+// continue asynchronously.
+type deletePodsOp struct {
+	// Must be "deletePods".
+	Opcode operationCode
+	// Namespace the pods should be deleted from.
+	Namespace string
+	// Labels used to filter the pods to delete.
+	Labels map[string]string
```
Comment (sanposhiho) [inline test/integration/scheduler_perf/scheduler_perf.go:673]: Actually, in which case will we need label selector? (isn't `Namespace` enough?)
If you're going to use it to delete Pods created by multiple `createPodsOp`, you need to mark `Namespace` as optional.

Comment (macsko) [reply test/integration/scheduler_perf/scheduler_perf.go]: Topology spreading plugin works only within a single namespace. We'll need the unschedulable pods blocked by topology spreading to be is the same namespace as the blocker pods that we'll need to delete.

File: test/integration/scheduler_perf/config/performance-config.yaml [hunk 2]
```diff
@@ -1333,7 +1333,10 @@
     # Create pods that will be gradually deleted after being scheduled.
   - opcode: createPods
     countParam: $deletingPods
+  - opcode: deletePods
```
Comment (alculquicondor) [inline test/integration/scheduler_perf/config/performance-config.yaml:1336]: can you improve the description of the test to give some high level idea of the intermediate steps?

**Review by alculquicondor — COMMENTED** [review]
/approve
Leaving LGTM to @sanposhiho

**Review by sanposhiho — COMMENTED** [review]
/lgtm

/hold
for the last comment from Aldo https://github.com/kubernetes/kubernetes/pull/127471/files#r1767326718

File: test/integration/scheduler_perf/config/performance-config.yaml (hunk 2 above)
Comment (macsko) [reply test/integration/scheduler_perf/config/performance-config.yaml:1339]: Done

File: test/integration/scheduler_perf/config/performance-config.yaml [hunk 3]
```diff
@@ -1402,7 +1408,10 @@
     countParam: $deletingPods
     podTemplatePath: config/templates/pod-with-finalizer.yaml
     skipWaitToCompletion: true
+  - opcode: deletePods
```
Comment (sanposhiho) [inline test/integration/scheduler_perf/config/performance-config.yaml:1411]: This one too.

**Review by sanposhiho — APPROVED** [review]
/lgtm
/approve
/unhold

**Pull request merged**